TypeCobol/Compiler/CodeElements/Expressions/QualifiedName.cs

```diff
@@ -85,26 +85,28 @@ public int IndexOf(string item)
 
         public override bool Equals(System.Object other)
         {
-            if (other == null) return false;
-            return Equals(other as QualifiedName);
+            return Equals(other as AbstractQualifiedName);
         }
-        public virtual bool Equals(QualifiedName other)
+        public virtual bool Equals(AbstractQualifiedName abstractQualifiedName)
         {
-            if (other == null) return false;
-            //			if (other.IsExplicit != IsExplicit) return false;
-            if (other.Count != Count) return false;
+            if (Object.ReferenceEquals(this, abstractQualifiedName)) return true;
+            if (Object.ReferenceEquals(null, abstractQualifiedName)) return false;
+
+            if (abstractQualifiedName.Count != Count) return false;
             for (int c = 0; c < Count; c++)
-                if (!other[c].Equals(this[c])) return false;
+                if (!abstractQualifiedName[c].Equals(this[c])) return false;
             return true;
         }
         public override int GetHashCode()
         {
-            int hash = 13;
-            //			hash = hash*7 + IsExplicit.GetHashCode();
-            hash = hash * 7 + Count.GetHashCode();
-            foreach (string part in this)
-                hash = hash * 7 + part.GetHashCode();
-            return hash;
+            unchecked
+            {
+                int hash = 13;
+                hash = (hash * 397) ^ Count;
+                foreach (string part in this)
+                    hash = (hash * 397) + part.GetHashCode();
```

Comment:
Why `+` and not `^`  ?

Reply:
fixed by using ^.
https://github.com/TypeCobolTeam/TypeCobol/pull/1587/commits/1be2e096b4f27bb53cd71184aacdcd855f430e29